### DISCUSSION TITLE
Exists expression in Uppaal

### DISCUSSION THREAD

**Discussion by PARNEET22** [UPPAALModelChecker/UPPAAL-Meta#199] [General]
How to use exist expression in inside a function.

I want a function inside which i will check a variable's value exist in the number mentioned in the array,
if yes it will. return the value of another variable.

**Comment by thorulf4** [comment]
The `exists` and `forall` keywords iterate over a range and evaluates an expression foreach value in said range.
`exists` returns true if just one of the expressions are true
`forall` return true if all expressions are true

More pratical example:
```
const int N = 5;
int array[N] = { 0, 1, 2, 3, 4};

bool array_contains_two(){
    return exists(i: int[0,N-1]) array[i] == 2;
}
```

`exists(i: int[0,N-1]) array[i] == 2;` lets decompose this expression

`(i: int[0,N-1])` says we'll use the variable `i` to iterate over the range `int[0,N-1]` i.e. from 0 to and including N-1

`array[i] == 2` this is the expression we check, in this case we check if an array element is 2. Because the range [0, N-1] is the size of array this will check every element in array.

I hope this helps

**Reply by PARNEET22** [reply]
Thank you.. it would be of great help
